fenix/app/src/main/java/org/mozilla/fenix/settings/SecretSettingsFragment.kt

```diff
@@ -28,6 +34,11 @@ class SecretSettingsFragment : PreferenceFragmentCompat() {
     }
 
     override fun onCreatePreferences(savedInstanceState: Bundle?, rootKey: String?) {
+        var debugDrawerEnabled = false
+        lifecycleScope.launch {
+            debugDrawerEnabled = context?.debugSettings?.debugDrawerEnabledFlow?.first() ?: false
```

Comment:
Do we want to collect the flow rather than just take the value out of it so that we get reactivity?

Reply:
I only need the one-time read, since the UI will update correctly with the user's input

Reply:
Is the read guaranteed to complete before we finish setting up the UI initially? Since we're launching a coroutine to collect the read in, it seems like we could reach the end up setup before `debugDrawerEnabled` has been updated the result

Reply:
I don't believe so, but we have a couple options here:
- Use `runBlocking` or our `runBlockingIncrement` for this singular read
- Pre-load the data into memory ([doc](https://developer.android.com/topic/libraries/architecture/datastore#synchronous))

For option 2, we just fire off pre-read of the store inside of onCreate, essentially.

Reply:
Seems like option 2 would be preferable based on the doc? 

I haven't tested this but throwing out there, could you wrap the preference in a `lifecycleScope.launch` such as:
```
 lifecycleScope.launch {
            val debugDrawerEnabled = context?.debugSettings?.isDebugDrawerEnabled?.first() ?: false

            requirePreference<SwitchPreference>(R.string.pref_key_enable_debug_drawer).apply {
                isVisible = Config.channel.isNightlyOrDebug
                isChecked = debugDrawerEnabled
                onPreferenceChangeListener =
                    Preference.OnPreferenceChangeListener { _, newValue ->
                        lifecycleScope.launch {
                            context.debugSettings.updateDebugDrawerEnabled(enabled = newValue as Boolean)
                        }
                        true
                    }
            }
        }
``` 
or will this have an issue drawing async?

Reply:
nit: I also wonder whether the `?.first() ?: false` should be move directly to `debugDrawerEnabledFlow`

Reply:
I'll give the preference wrap a try.

I'll move it into the data store file and use `false` as a default static value